src/dotnet/Chat/Markup/Visitors/MarkupTrimmer.cs

```diff
@@ -74,7 +74,7 @@ protected override Markup VisitCodeBlock(CodeBlockMarkup markup)
         if (sb.Length == 0)
             return AppendEnd();
         Append(sb.Length);
-        return new MarkupSeq(markup with { Code = sb.ToString() }, AppendEnd());
+        return new MarkupSeq(markup with { Code = sb.ToString() });
```

Comment:
You shouldn't remove `AppendEnd` call here - see the first `if`, we need to append end in this case, coz we're trimming the code fragment.

Reply:
fixed